src/lib.rs

```diff
@@ -750,6 +756,38 @@ impl Gdbm {
             .into_iter()
             .try_for_each(|(offset, length)| self.free_record(offset, length))
     }
+
+    // API: compare_and_swap
+    pub fn compare_and_swap(
+        &mut self,
+        key: Vec<u8>,
+        old: Option<&[u8]>,
+        new: Option<Vec<u8>>,
+    ) -> io::Result<CompareAndSwapSummary> {
+        let current = self.get(&key)?;
+
+        if old != current.as_deref() {
```

Comment:
Help me understand this specific line of code.  ELI5.  :)

I would understand "if old ==" but I do not understand "if old !="

Reply:
We only swap if `old` (our parameter) isn't the same as `current` (the value in the database). That's the compare bit of compare and swap, at least as far as I understood from [sled's documentation](https://docs.rs/sled/latest/sled/struct.Db.html#method.compare_and_swap).

Reply:
Haha @jgarzik , hopefully you've seen sense and decided not to merge - this pattern sucks :)